pandas/core/missing.py

```diff
@@ -225,6 +227,56 @@ def find_valid_index(how: str, is_valid: npt.NDArray[np.bool_]) -> int | None:
     return idxpos  # type: ignore[return-value]
 
 
+def infer_limit_direction(limit_direction, method):
```

Comment:
Could we type this function?

Reply:
Doing so caused some trouble in part bc the NDFrame.interpolate annotation is inaccurate.  ive got a interpolate-cleanup branch that im hoping improves the situation a bit, but the real solution is #53607